From 02b29391258ad1a5520a11af0d708108a5228610 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Thu, 5 Mar 2020 14:36:23 +0200
Subject: [PATCH] speed up (and fix lags) in Q recents v3

	modified:   packages/SystemUI/legacy/recents/res/anim/recents_fast_toggle_app_home_exit.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_enter.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_exit.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_enter.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_exit.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_bounce.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_source.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_target.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_bounce.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_source.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_target.xml
	modified:   packages/SystemUI/legacy/recents/res/anim/recents_to_launcher_enter.xml
---
 .../res/anim/recents_fast_toggle_app_home_exit.xml   |  2 +-
 .../recents/res/anim/recents_from_launcher_enter.xml |  2 +-
 .../recents/res/anim/recents_from_launcher_exit.xml  |  2 +-
 .../recents/res/anim/recents_from_unknown_enter.xml  |  2 +-
 .../recents/res/anim/recents_from_unknown_exit.xml   |  2 +-
 .../recents_launch_next_affiliated_task_bounce.xml   | 12 ++++++------
 .../recents_launch_next_affiliated_task_source.xml   |  4 ++--
 .../recents_launch_next_affiliated_task_target.xml   |  2 +-
 .../recents_launch_prev_affiliated_task_bounce.xml   |  6 +++---
 .../recents_launch_prev_affiliated_task_source.xml   |  2 +-
 .../recents_launch_prev_affiliated_task_target.xml   |  4 ++--
 .../recents/res/anim/recents_to_launcher_enter.xml   |  2 +-
 12 files changed, 21 insertions(+), 21 deletions(-)

diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_fast_toggle_app_home_exit.xml b/packages/SystemUI/legacy/recents/res/anim/recents_fast_toggle_app_home_exit.xml
index 69edcc757ba..452870a53bc 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_fast_toggle_app_home_exit.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_fast_toggle_app_home_exit.xml
@@ -23,5 +23,5 @@
   <alpha android:fromAlpha="1.0" android:toAlpha="1.0"
          android:fillEnabled="true"
          android:fillBefore="true" android:fillAfter="true"
-         android:duration="250"/>
+         android:duration="125"/>
 </set>
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_enter.xml b/packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_enter.xml
index 00b3dfda135..dd0e4eb94e3 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_enter.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_enter.xml
@@ -24,5 +24,5 @@
          android:fillEnabled="true"
          android:fillBefore="true" android:fillAfter="true"
          android:interpolator="@android:interpolator/linear_out_slow_in"
-         android:duration="150"/>
+         android:duration="75"/>
 </set>
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_exit.xml b/packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_exit.xml
index 33831b8c0a3..00980027175 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_exit.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_from_launcher_exit.xml
@@ -24,5 +24,5 @@
          android:fillEnabled="true"
          android:fillBefore="true" android:fillAfter="true"
          android:interpolator="@interpolator/recents_from_launcher_exit_interpolator"
-         android:duration="133"/>
+         android:duration="66"/>
 </set>
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_enter.xml b/packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_enter.xml
index da1dee00754..088e021c5d1 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_enter.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_enter.xml
@@ -24,5 +24,5 @@
          android:fillEnabled="true"
          android:fillBefore="true" android:fillAfter="true"
          android:interpolator="@android:interpolator/linear"
-         android:duration="200"/>
+         android:duration="100"/>
 </set>
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_exit.xml b/packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_exit.xml
index 31cf26a9fdf..0d7eaf06ed1 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_exit.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_from_unknown_exit.xml
@@ -24,5 +24,5 @@
          android:fillEnabled="true"
          android:fillBefore="true" android:fillAfter="true"
          android:interpolator="@android:interpolator/fast_out_slow_in"
-         android:duration="200"/>
+         android:duration="100"/>
 </set>
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_bounce.xml b/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_bounce.xml
index 74f2814b2ce..911c1a9d6a2 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_bounce.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_bounce.xml
@@ -24,26 +24,26 @@
     <translate android:fromYDelta="0" android:toYDelta="2%"
         android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
         android:interpolator="@android:interpolator/fast_out_slow_in"
-        android:duration="133"/>
+        android:duration="66"/>
 
     <scale android:fromXScale="1.0" android:toXScale="0.98"
         android:fromYScale="1.0" android:toYScale="0.98"
         android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
         android:pivotX="50%p" android:pivotY="50%p"
         android:interpolator="@android:interpolator/fast_out_slow_in"
-        android:duration="133" />
+        android:duration="66" />
 
     <translate android:fromYDelta="0" android:toYDelta="-2%"
         android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
         android:interpolator="@interpolator/recents_launch_prev_affiliated_task_bounce_ydelta"
-        android:startOffset="133"
-        android:duration="217"/>
+        android:startOffset="66"
+        android:duration="108"/>
 
     <scale android:fromXScale="1.0" android:toXScale="1.02040816326531"
         android:fromYScale="1.0" android:toYScale="1.02040816326531"
         android:fillEnabled="true" android:fillBefore="false" android:fillAfter="true"
         android:pivotX="50%p" android:pivotY="50%p"
         android:interpolator="@interpolator/recents_launch_next_affiliated_task_bounce_scale"
-        android:startOffset="133"
-        android:duration="217" />
+        android:startOffset="66"
+        android:duration="108" />
 </set>
\ No newline at end of file
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_source.xml b/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_source.xml
index f0fd6845880..96f7904254b 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_source.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_source.xml
@@ -23,12 +23,12 @@
     <alpha android:fromAlpha="1.0" android:toAlpha="0.6"
         android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
         android:interpolator="@android:interpolator/accelerate_cubic"
-        android:duration="150"/>
+        android:duration="75"/>
 
     <scale android:fromXScale="1.0" android:toXScale="0.9"
         android:fromYScale="1.0" android:toYScale="0.9"
         android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
         android:pivotX="50%p" android:pivotY="50%p"
         android:interpolator="@android:interpolator/fast_out_slow_in"
-        android:duration="300" />
+        android:duration="150" />
 </set>
\ No newline at end of file
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_target.xml b/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_target.xml
index 170ac829c3c..3df68c6f31a 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_target.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_launch_next_affiliated_task_target.xml
@@ -24,5 +24,5 @@
                android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
                android:interpolator="@android:interpolator/decelerate_quint"
                android:startOffset="50"
-               android:duration="250" />
+               android:duration="125" />
 </set>
\ No newline at end of file
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_bounce.xml b/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_bounce.xml
index b19167da3dd..d70c6be41f5 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_bounce.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_bounce.xml
@@ -23,11 +23,11 @@
     <translate android:fromYDelta="0%" android:toYDelta="10%"
                android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
                android:interpolator="@android:interpolator/fast_out_slow_in"
-               android:duration="133" />
+               android:duration="66" />
 
     <translate android:fromYDelta="0%" android:toYDelta="-10%"
                android:fillEnabled="true" android:fillBefore="false" android:fillAfter="true"
                android:interpolator="@interpolator/recents_launch_prev_affiliated_task_bounce_ydelta"
-               android:startOffset="133"
-               android:duration="217" />
+               android:startOffset="66"
+               android:duration="108" />
 </set>
\ No newline at end of file
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_source.xml b/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_source.xml
index ad5341bf8ff..31f29d44ca3 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_source.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_source.xml
@@ -23,5 +23,5 @@
     <translate android:fromYDelta="0%" android:toYDelta="110%"
                android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
                android:interpolator="@android:interpolator/accelerate_quint"
-               android:duration="300" />
+               android:duration="150" />
 </set>
\ No newline at end of file
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_target.xml b/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_target.xml
index 7687f0286f2..4d15505a28a 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_target.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_launch_prev_affiliated_task_target.xml
@@ -24,7 +24,7 @@
         android:fillEnabled="true" android:fillBefore="true" android:fillAfter="true"
         android:interpolator="@android:interpolator/decelerate_cubic"
         android:startOffset="75"
-        android:duration="150"/>
+        android:duration="75"/>
 
     <scale android:fromXScale="0.9" android:toXScale="1.0"
         android:fromYScale="0.9" android:toYScale="1.0"
@@ -32,5 +32,5 @@
         android:interpolator="@android:interpolator/linear_out_slow_in"
         android:pivotX="50%p" android:pivotY="50%p"
         android:startOffset="75"
-        android:duration="225" />
+        android:duration="112" />
 </set>
\ No newline at end of file
diff --git a/packages/SystemUI/legacy/recents/res/anim/recents_to_launcher_enter.xml b/packages/SystemUI/legacy/recents/res/anim/recents_to_launcher_enter.xml
index 544ec88d2bf..09f6103e9ca 100644
--- a/packages/SystemUI/legacy/recents/res/anim/recents_to_launcher_enter.xml
+++ b/packages/SystemUI/legacy/recents/res/anim/recents_to_launcher_enter.xml
@@ -24,5 +24,5 @@
          android:fillEnabled="true"
          android:fillBefore="true" android:fillAfter="true"
          android:interpolator="@interpolator/recents_to_launcher_enter_interpolator"
-         android:duration="133"/>
+         android:duration="66"/>
 </set>
-- 
2.17.1

